API/gimvicurnik/__init__.py

```diff
@@ -373,8 +372,36 @@ def _get_documents():
                     for model in query
                 ]
             )
+        @self.app.route("/circular/atom")
+        def _get_atom():
+            query = self.session.query(Document.date, Document.type, Document.url, Document.description).filter(or_(Document.type == "circular", Document.type == "None")).order_by(Document.date)
+
+            config = self.config["sources"]["eclassroom"]["pluginfile"]
+            token = self.config["sources"]["eclassroom"]["token"]
 
+            return render_template("atom.xml", entries = query, last_updated = max(model.date for model in query).strftime("%Y-%m-%d"))
+        @self.app.route("/circular/rss")
+        def _get_rss():
+            query = self.session.query(Document.date, Document.type, Document.url, Document.description).filter(or_(Document.type == "circular", Document.type == "None")).order_by(Document.date)
+
+            config = self.config["sources"]["eclassroom"]["pluginfile"]
+            token = self.config["sources"]["eclassroom"]["token"]
 
+            return render_template("rss.xml", entries = query)
+        @self.app.route("/calendar/classes/<string:class_>")
+        def _get_calendar_for_classes(class_):
+            h = {
+                0 : ("071000", "075500"),
+                1 : ("080000", "084500"),
+                2 : ("085000", "093500"),
+                3 : ("094000", "102500"),
+                4 : ("105500", "114000"),
+                5 : ("114500", "123000"),
+                6 : ("123500", "132000"),
+                7 : ("132500", "141000"),
+                8 : ("141500", "150000")
+            }
+            return render_template("sample.ics", nadomescanja = Class.get_icsfeed(self.session, [class_]), h = h)
```

Comment:
Za koledar pa:

* `/calendars/timetable/*` (npr. `/calendars/timetable/classes/2A`), ki dela isto kot `/timetable` URLji (samo urnik brez nadomeščanj, če se da ponavljanje eventov tedensko), samo da je v ICAL
* `/calendars/substitutions/*` (npr. `/calendars/substitutions/classes/2A`), ki je pa podoben kot `/substitutions` ampak brez datuma in samo vrne vsa nadomeščanja za katera ve
* `/calendars/schedule/*` (npr. `/calendars/schedule/classes/2A`), ki je pa podoben kot `/schedule` ampak brez datuma in samo vrne vse razporede za katere ve

Približno tko kot obstoječi URLji samo da je v iCalendar.
Ena verzija tako da da podatke za vse razrede, ena pa samo za tistega, za katerega je request (tko kot je tudi za obstoječe URLje).

Reply:
>>> Ena verzija tako da da podatke za vse razrede,
But why?

Reply:
Ne vem. Mogoče.